iris_grib/tests/integration/round_trip/test_grid_definition_section.py

```diff
@@ -0,0 +1,140 @@
+# Copyright iris-grib contributors
+#
+# This file is part of iris-grib and is released under the LGPL license.
+# See COPYING and COPYING.LESSER in the root of the repository for full
+# licensing details.
+"""
+Integration test for round-trip loading and saving of various grids.
+
+"""
+
+# import iris_grib.tests first so that some things can be initialised
+# before importing anything else.
+import iris_grib.tests as tests
+
+from iris import load_cube, save
+from iris.coord_systems import RotatedGeogCS
+from iris.fileformats.pp import EARTH_RADIUS as UM_DEFAULT_EARTH_RADIUS
+from iris.util import is_regular
+
+from iris.tests import TestGribMessage
+
+from iris_grib.grib_phenom_translation import GRIBCode
+
+
+class TestGDT5(TestGribMessage):
+    def test_save_load(self):
+        # Load sample UKV data (variable-resolution rotated grid).
+        path = tests.get_data_path(("PP", "ukV1", "ukVpmslont.pp"))
+        cube = load_cube(path)
+
+        # Extract a single 2D field, for simplicity.
+        self.assertEqual(cube.ndim, 3)
+        self.assertEqual(cube.coord_dims("time"), (0,))
+        cube = cube[0]
+
```

Comment:
This cube loads in with a shape of (928, 744).

6 years ago, eccodes had a bug such that it could only save square data. Thus, in Iris this tests square off the data before saving it (see [here](https://github.com/SciTools/iris/blob/836244ea92d4dff32e411be329524234a5f9f6dd/lib/iris/tests/integration/test_grib2.py#L180)). I have removed that as the bug has been fixed (many years ago!)